finger/admin.py

```diff
@@ -1,5 +1,34 @@
 from django.contrib import admin
 from .models import User
+from django.db.models import Q
+from datetime import datetime, timedelta
+
+class MemberStatusListFilter(admin.SimpleListFilter):
+    title = "Members Status"
+    parameter_name = 'member_status'
+
+    def lookups(self, request, model_admin):
+        return (
+            ('active', "Active"),
+            ('recent', "Recent"),
+        )
+
+    def queryset(self, request, queryset):
+        if self.value() == 'active' or self.value() == 'recent':
+
+            if self.value() == 'active':
+                t = datetime.now()
+            else:
+                t = datetime.now() - timedelta(days=2*365)
+
+            if t.month <= 7:
+                ths_member_verified = Q(ths_verified_vt__gte=t.year)
+            else:
+                ths_member_verified = Q(ths_verified_ht__gte=t.year)
+
+            return queryset.filter(
+                Q(payed_year__gte=t.year) | ths_member_verified
+            ).exclude(is_active__exact=False)
```

Comment:
Honorary members should probably count as "active". If they show up on a meeting, they can vote. If a list of non-honorary active members is wanted, they can be included by using the two filters simultaneous (different filters are and:ed, not or:ed, in the gui).

Reply:
Agree, I have a commit that fixes that.